core/amber/src/test/scala/edu/uci/ics/texera/workflow/operators/specializedFilter/SpecializedFilterOpExecSpec.scala

```diff
@@ -0,0 +1,226 @@
+package edu.uci.ics.texera.workflow.operators.specializedFilter
+
+import edu.uci.ics.amber.engine.common.InputExhausted
+import edu.uci.ics.amber.engine.common.virtualidentity.{LayerIdentity, LinkIdentity}
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, AttributeType, Schema}
+import edu.uci.ics.texera.workflow.operators.filter.{
+  ComparisonType,
+  FilterPredicate,
+  SpecializedFilterOpDesc,
+  SpecializedFilterOpExec
+}
+import org.scalatest.BeforeAndAfter
+import org.scalatest.flatspec.AnyFlatSpec
+
+import java.util.Arrays.asList
+
+class SpecializedFilterOpExecSpec extends AnyFlatSpec with BeforeAndAfter {
+  var counter: Int = 0
+
+  val tupleSchema: Schema = Schema
+    .newBuilder()
+    .add(new Attribute("string", AttributeType.STRING))
+    .add(new Attribute("int", AttributeType.INTEGER))
+    .add(new Attribute("bool", AttributeType.BOOLEAN))
+    .add(new Attribute("long", AttributeType.LONG))
+    .build()
+
+  def layerID(): LayerIdentity = {
+    counter += 1
+    LayerIdentity("" + counter, "" + counter, "" + counter)
+  }
+
+  def linkID(): LinkIdentity = LinkIdentity(layerID(), layerID())
+
+  def markerValue(): String = {
+    "RandomValue"
+  }
+
+  def allNullTuple(): Tuple = {
+    counter += 1
+    Tuple
+      .newBuilder(tupleSchema)
+      .add(new Attribute("string", AttributeType.STRING), null)
+      .add(new Attribute("int", AttributeType.INTEGER), null)
+      .add(new Attribute("bool", AttributeType.BOOLEAN), null)
+      .add(new Attribute("long", AttributeType.LONG), null)
+      .build()
+  }
+
+  def nonNullTuple(): Tuple = {
+    counter += 1
+    Tuple
+      .newBuilder(tupleSchema)
+      .add(new Attribute("string", AttributeType.STRING), "hello")
+      .add(new Attribute("int", AttributeType.INTEGER), 0)
+      .add(new Attribute("bool", AttributeType.BOOLEAN), false)
+      .add(new Attribute("long", AttributeType.LONG), Long.MaxValue)
+      .build()
+  }
+
+  before {
+    counter = 0
+  }
+
+  it should "open and close" in {
+    val opExec = new SpecializedFilterOpExec(new SpecializedFilterOpDesc())
+    opExec.open()
+    opExec.close()
+  }
+
+  it should "include all tuples from 1 completely null input stream of string type for is null" in {
```

Comment:
it should "include null tuples when filter is is_null" sounds more comprehensive?

Reply:
yep!